tests/generator-Tests/Unit-Tests/GenBaseTests.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using MonoDroid.Generation;
+using NUnit.Framework;
+
+namespace generatortests
+{
+	[TestFixture]
+	public class GenBaseTests
+	{
+		CodeGenerationOptions options = new CodeGenerationOptions ();
+
+		[Test]
+		public void PropertyRequiresNew ()
+		{
+			var c = SupportTypeBuilder.CreateClassWithProperty ("MyClass", "java.myClass", "Handle", "int", options);
+			Assert.True (c.RequiresNew (c.Properties.First ()));
+
+			c.Properties.First ().Name = "GetHashCode";
+			Assert.True (c.RequiresNew (c.Properties.First ()));
+
+			c.Properties.First ().Name = "GetType";
+			Assert.True (c.RequiresNew (c.Properties.First ()));
+
+			c.Properties.First ().Name = "ToString";
+			Assert.True (c.RequiresNew (c.Properties.First ()));
+
+			c.Properties.First ().Name = "Equals";
+			Assert.True (c.RequiresNew (c.Properties.First ()));
+
+			c.Properties.First ().Name = "ReferenceEquals";
+			Assert.True (c.RequiresNew (c.Properties.First ()));
+
+			c.Properties.First ().Name = "Handle2";
```

Comment:
I don't understand why `Handle2` should require a `new`.

Reply:
It's `Assert`ing that it doesn't require a `new`.  :)